### ISSUE TITLE
How to extract the texture information and save it as an image

### ISSUE THREAD

**Issue by Koushik0901** [sicxu/Deep3DFaceRecon_pytorch#34]
Hi there, I am new to this stuff... So it would be nice if you could guide me on how to save the texture information of the mesh as png file

**Comment by elenacliu** [comment]
I have the same problem with you. The output of this model is .obj files, however the common 3D modeling software takes (obj, mtl, png) tuples as inputs, which contain texture information in png images. Although the .obj outputs have texture information in vertices information, the software seems to ignore the (R, G, B) values of these vertices.

**Comment by Koushik0901** [comment]
Yeah, you have to perform texture baking on blender to get the texture as png

**Comment by elenacliu** [comment]
@Koushik0901 I'm sorry, but I found that when Blender imports an obj file, it ignores vertex color, so can I get the texture image when I perform texture baking?

https://docs.blender.org/manual/en/latest/addons/import_export/scene_obj.html

![图片](https://user-images.githubusercontent.com/44845151/148728391-29d7c5cc-3128-4ded-a728-c9b22001d3e3.png)

**Comment by Koushik0901** [comment]
> @Koushik0901 I'm sorry, but I found that when Blender imports an obj file, it ignores vertex color, so can I get the texture image when I perform texture baking?
> 
> https://docs.blender.org/manual/en/latest/addons/import_export/scene_obj.html
> 
> ![图片](https://user-images.githubusercontent.com/44845151/148728391-29d7c5cc-3128-4ded-a728-c9b22001d3e3.png)

Yeah, You can get the texture(png file) by performing texture baking

**Comment by elenacliu** [comment]
> I have the same problem with you. The output of this model is .obj files, however the common 3D modeling software takes (obj, mtl, png) tuples as inputs, which contain texture information in png images. Although the .obj outputs have texture information in vertices information, the software seems to ignore the (R, G, B) values of these vertices.

@Koushik0901 

Ah, I find that although official implementation of [import_obj addons](https://github.com/blender/blender-addons/blob/master/io_scene_obj/import_obj.py) does not support parsing .obj vertex color, you can

1. add parsing codes yourself
2. convert obj to ply via meshlab or sth similar

After converting obj to ply, you can generate texture map from vertex color by this script:

https://github.com/jiegec/blender-scripts
